core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1644,13 +1672,53 @@ class KafkaZkClient private (zooKeeperClient: ZooKeeperClient, isSecure: Boolean
       }
     }
 
+    // This method is part of the work around done in the KAFKA-7165, once ZOOKEEPER-2985 is complete, this code must
+    // be deleted.
+    private def delete(): Code = {
+      val deleteRequest = DeleteRequest(path, ZkVersion.MatchAnyVersion)
+      val deleteResponse = retryRequestUntilConnected(deleteRequest)
+      deleteResponse.resultCode match {
+        case code@ Code.OK => code
```

Comment:
Could we just move the call to updateCurrentZKSessionId() here? This way, there will be a single place where updateCurrentZKSessionId() will be called.

Reply:
Hmmmm, where exactly @junrao?

We are setting the `currentZooKeeperSessionId` at the end of `private def checkedEphemeralCreate(path: String, data: Array[Byte])` method, this ensures that the first time the broker tries to registered we have the session Id used.

If we do not do that, the method `isZKSessionTheEphemeralOwner` will return `false` (since `-1 != zooKeeperClient.sessionId`) and will not try to recreate the ephemeral node.

Then we are setting `currentZooKeeperSessionId` again if the condition `case Code.OK if shouldReCreateEphemeralZNode(ephemeralOwnerId) =>` applies, becase this will update the `currentZooKeeperSessionId` with the newest session, useful when trying to recreate the ephemeral node.

Am pretty sure am missing something since you point this out, but I can not see it, hope you can give me a clue :)

Reply:
Sorry, I didn't make it clear. Basically, we want to set the session id after a successful creation of the ephemeral path. I was thinking  that we can call updateCurrentZKSessionId() after line 1667 in CheckedEphemeral.create() when the return code is ok. Then, we don't need to call updateCurrentZKSessionId() anywhere else.

Reply:
+1